### DISCUSSION TITLE
W/libc: Unable to set property "debug.sf.dequeuebuffer" to "1": connection failed; errno=13

### DISCUSSION THREAD

**Discussion by mansourmoufid** [android/ndk#1734] [General]
I am using the camera2 API.

When my library exports the symbol "dump_buffer", ACameraCaptureSession fails and onCaptureFailed is called with the reason CAPTURE_FAILURE_REASON_ERROR.

In logcat I see this:

    W/libc: Unable to set property "debug.sf.dequeuebuffer" to "1": connection failed; errno=13 (Permission denied)
    E/BufferQueueProducer: [ImageReader-1088x1088f23u3m1-22596-0](id:584400000000,api:4,p:823,c:22596) waitForFreeSlotThenRelock: timeout

I can reliably trigger this by adding or removing "dump_buffer":

    __attribute__((visibility(default))) void dump_buffer(void) {}

The solution is to just remove or rename "dump_buffer".

Leaving this here in case anyone else stumbles on it. The cause is not obvious from the errors (I figured it out by total accident).

**Comment by mansourmoufid** [comment]
cc @[ChathuragaShan](https://github.com/ChathuragaShan) (https://stackoverflow.com/questions/67145610/camera-2-preview-freezes-when-i-uses-yuv-420-888-image-reader-as-a-target-surfac)

cc @[trueinnidhi](https://github.com/trueinnidhi) (https://github.com/flutter/flutter/issues/94232)

**Comment by enh-google** [comment]
`debug.sf.dequeuebuffer` isn't referenced in AOSP or internal master... so this is either something in your code (if you're an OEM/SoC vendor), or something specific to whatever device you're running on?
